core/src/main/kotlin/com/malinskiy/marathon/execution/DevicePoolActor.kt

```diff
@@ -101,6 +100,8 @@ class DevicePoolActor(private val poolId: DevicePoolId,
     private suspend fun executeBatch(device: DeviceInfo, batch: TestBatch) {
         devices[device.serialNumber]?.run {
             safeSend(DeviceEvent.Execute(batch))
+        } ?: run {
+            queue.send(QueueMessage.ReturnBatch(device, batch))
```

Comment:
`queue.send` → `queue.safeSend` to avoid rare, but unfortunate exceptions

Reply:
Fixed in all places of that class